lib/parser.js

```diff
@@ -27,68 +27,52 @@ THE SOFTWARE
 var di = require('di');
 
 module.exports = ParserFactory;
+
 di.annotate(ParserFactory, new di.Provide('DHCP.parser'));
 di.annotate(ParserFactory, new di.Inject('DHCP.protocol', 'Logger', 'Assert'));
-function ParserFactory(protocol, Logger, assert) {
-    var logger = Logger.initialize(ParserFactory);
-    var parser = {};
 
-    parser.parse = function(msg) {
-        var len;
-        //logger.info(rinfo.address + ':' + rinfo.port + '/' + msg.length + 'b');
-        var packet = {
-            op: protocol.BOOTPMessageType.get(msg.readUInt8(0)),
-            // htype is combined into chaddr field object
-            hlen: msg.readUInt8(2),
-            hops: msg.readUInt8(3),
-            xid: msg.readUInt32BE(4),
-            secs: msg.readUInt16BE(8),
-            flags: msg.readUInt16BE(10),
-            ciaddr: readIpRaw(msg, 12),
-            yiaddr: readIpRaw(msg, 16),
-            siaddr: readIpRaw(msg, 20),
-            giaddr: readIpRaw(msg, 24),
-            chaddr: protocol.createHardwareAddress(
-                protocol.ARPHardwareType.get(msg.readUInt8(1)),
-                readAddressRaw(msg, 28, msg.readUInt8(2))),
-            sname: trimNulls(msg.toString('ascii', 44, 108)),
-            fname: trimNulls(msg.toString('ascii', 108, 236)),
-            magic: msg.readUInt32BE(236),
-            options: {}
-        };
-        var offset = 240;
-        var code = 0;
+function ParserFactory(protocol, Logger, assert) {
+    return {
+        logger: Logger.initialize(ParserFactory),
+        len: null,
```

Comment:
`len` is a property that is used by some of the parser methods. It is being assigned to null because the code before didn't set a value of len when it was declared.

By declaring `len` a property of the parser object in the literal definition it simplifies v8's job of determining its hidden type. See https://developers.google.com/v8/design?hl=en

If `len` was declared dynamically it may not optimize the class as efficiently.